floor_generator/lib/misc/foreign_key_action.dart

```diff
@@ -18,9 +16,21 @@ abstract class ForeignKeyAction {
         return 'SET DEFAULT';
       case ForeignKeyAction.cascade:
         return 'CASCADE';
-      case ForeignKeyAction.noAction:
       default:
-        return 'NO ACTION';
+        assert(false, 'Unexpected enum case $this');
```

Comment:
`assert`s only are enabled in debug mode. Was this intentional? https://dart.dev/guides/language/language-tour#assert

Reply:
I wanted to throw an Error without overhead (if anything ever throws this error, it is our fault and not the one of library users)... assert seems to be the wrong way then... Do you have an idea?

Reply:
As far as I know, `assert` throws an `AssertError`. So in the end, it's just an `Error`.
I've started using the built-in errors such as `UnsupportedError`, `StateError` or `ArgumentError` recently, which I like.